Expensify/App
# Refactor the loading state of the sign on forms

### Details
This refactors the forms to use an Onyx state property rather than their individual internal state

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/149339
Fixes https://github.com/Expensify/Expensify.cash/issues/1133
Fixes https://github.com/Expensify/Expensify.cash/issues/1131

### Tests
1. Make sure you're signed out of the app
2. Sign in with an existing account
3. Verify the form buttons in the sign-in flow have a proper loading state
3. Verify that each step except for the login step has a link for "Change Expensify login" and that clicking the link takes you back to the first step
3. Sign out
4. Sign in with a brand new account and verify the form buttons in the sign-in flow have a proper loading state
5. Sign out
6. Enter an email with an existing account and click continue
7. Navigate to `/#/setpassword/test`
8. Enter the password `123` and submit the form
9. Verify the button shows a brief loading state and there is an error that the password is invalid
10. Enter the password `asdfASDF00` and submit the form
11. Verify there is an error message saying the validate code is likely expired (that's OK, because we used a fake validate code in the URL)

### Tested On


### Screenshots
<Add any necessary screenshots for all platforms if your change added or updated UI>

#### Web
no changes

#### Mobile Web
no changes

#### Desktop
no changes

#### iOS
no changes

#### Android
no changes
